x-pack/winlogbeat/module/routing/ingest/routing.yml

```diff
@@ -6,16 +6,16 @@ processors:
       value: '{{_ingest.timestamp}}'
   - pipeline:
       name: '{< IngestPipeline "security" >}'
-      if: ctx?.winlog?.channel == 'Security' && ['Microsoft-Windows-Eventlog', 'Microsoft-Windows-Security-Auditing'].contains(ctx?.winlog?.provider_name)
+      if: ctx.winlog?.channel instanceof String && ctx.winlog.channel.toLowerCase() == 'security' && ['Microsoft-Windows-Eventlog', 'Microsoft-Windows-Security-Auditing'].contains(ctx.winlog?.provider_name)
```

Comment:
Any pipeline test for this change?

Reply:
The changes are tested in the [integration PR](https://github.com/elastic/integrations/pull/8242) as noted above. The issue with testing here is that we need an evtx with the case difference. The system and pipeline tests here test for regression already though.